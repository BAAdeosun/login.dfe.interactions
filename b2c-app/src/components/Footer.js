import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
