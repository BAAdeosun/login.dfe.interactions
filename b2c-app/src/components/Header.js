import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;