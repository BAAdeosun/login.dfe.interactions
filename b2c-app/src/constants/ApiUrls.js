export const API_URLS = {
    //TODO replace this URL with placeholder __--changeEmailAPI--__ when env vars are ready
    CHANGE_EMAIL: "https://dev-api-customerengagement.platform.education.gov.uk/authncs/ChangeEmail"
};