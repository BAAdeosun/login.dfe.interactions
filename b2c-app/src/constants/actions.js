export const ACTIONS = {
    SIGNUP: 'signup',
    LOGIN: 'login',
    RESET_PASSWORD: 'reset-password',
    FIND_EMAIL: 'find-email',
    CHANGE_EMAIL: 'change-email'
};