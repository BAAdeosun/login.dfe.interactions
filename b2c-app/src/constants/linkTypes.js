export const LINK_TYPES = {
    LINK: 'link',
    BUTTON: 'button',
    API_CALL: 'api'
};