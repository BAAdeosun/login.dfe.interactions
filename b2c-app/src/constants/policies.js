export const POLICIES = {
    SIGNIN_INVITATION: 'B2C_1A_Signin_Invitation',
    ACCOUNT_SIGNUP: 'B2C_1A_Account_Signup',
    PASSWORD_RESET: 'B2C_1A_Password_Reset',
    PASSWORD_RESET_CONFIRMATION: 'B2C_1A_Password_Reset_Confirmation',
    FIND_EMAIL: 'B2C_1A_Find_Email',
    SIGNUP_CONFIRMATION: 'B2C_1A_Signup_Confirmation',
    SIGNUP_INVITATION: 'B2C_1A_Signup_Invitation',
    CHANGE_EMAIL: 'B2C_1A_Change_Email'
};