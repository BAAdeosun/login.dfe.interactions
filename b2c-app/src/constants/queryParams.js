export const QUERY_PARAMS = {
    CLIENT_ID: 'client_id',
    REDIRECT_URI: 'redirect_uri',
    ID_TOKEN_HINT: 'id_token_hint',
    TENANT_ID: 'tenand_id',
    POLICY: 'policy',
    PAGE: 'custom_ui_page'
}