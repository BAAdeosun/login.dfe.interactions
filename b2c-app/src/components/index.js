import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}