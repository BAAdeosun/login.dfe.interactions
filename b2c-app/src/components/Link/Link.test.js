import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { LINK_TYPES } from '../../constants/linkTypes';
import { create, act } from 'react-test-renderer';

import components from '..';


it('renders without crashing', () => {
    shallow(<components.Link />);
});

describe('when URL passed in as prop', () => {

    it('renders correctly without specified type', () => {
        let tree;
        act(() => {
            tree = create(<components.Link url='https://test.com' />)
        });
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('renders correctly link of type button', () => {
        let tree;
        act(() => {
            tree = create(<components.Link type={LINK_TYPES.BUTTON} url='https://test.com' />)
        });
        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe('when URL is not specified', () => {

    it('renders correctly without specified type', () => {
        const tree = renderer
            .create(<components.Link />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly link of type button', () => {
        const tree = renderer
            .create(<components.Link type={LINK_TYPES.BUTTON} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});