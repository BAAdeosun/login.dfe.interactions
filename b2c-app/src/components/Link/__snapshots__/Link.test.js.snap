// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when URL is not specified renders correctly link of type button 1`] = `
<a
  className="govuk-button govuk-button--start"
  data-module="govuk-button"
  draggable="false"
  onClick={null}
  role="button"
/>
`;

exports[`when URL is not specified renders correctly without specified type 1`] = `
<a
  className="govuk-link"
  onClick={null}
/>
`;

exports[`when URL passed in as prop renders correctly link of type button 1`] = `
<a
  className="govuk-button govuk-button--start"
  data-module="govuk-button"
  draggable="false"
  href="https://test.com"
  onClick={null}
  role="button"
/>
`;

exports[`when URL passed in as prop renders correctly without specified type 1`] = `
<a
  className="govuk-link"
  href="https://test.com"
  onClick={null}
/>
`;
